src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -350,7 +350,14 @@ public ProgrammingExerciseStudentParticipation resumeProgrammingExercise(Program
         participation = configureBuildPlan(participation);
         // Note: the repository webhook (step 1c) already exists, so we don't need to set it up again, the empty commit hook (step 2c) is also not necessary here
         // and must be handled by the calling method in case it would be necessary
-        participation.setInitializationState(INITIALIZED);
+
+        // If the instructor retests a participation after the deadline set the state back to finished. Otherwise, the participation is initialized
```

Comment:
Explain why this doesn't affect test runs in the comment.

Reply:
An exercise in an exam does not have a deadline on its own. So the comment already describes the behavior correctly.